File: client/unity/Assets/Scripts/Assembly-CSharp/BaucuaMsgHandler.cs
public class BaucuaMsgHandler : IMiniGameMsgHandler
{
	public static BaucuaMsgHandler me;

	public static void init()
	{
		me = new BaucuaMsgHandler();
	}

	public static void onHandler()
	{
		BoardScr.numPlayer = 5;
		BoardListOnScr.type = BoardListOnScr.STYLE_5PLAYER;
		if (onMainMenu.isOngame)
		{
			RoomListOnScr.setName(3, BCBoardScr.gI());
		}
		else
		{
			RoomListOnScr.setName((Canvas.iOpenOngame != 0) ? 1 : 3, BCBoardScr.gI());
		}
		if (me == null)
		{
			me = new BaucuaMsgHandler();
		}
		CasinoMsgHandler.me.miniGameMessageHandler = me;
	}

	public void onMessage(Message msg)
	{
		sbyte b = msg.reader().readByte();
		sbyte b2 = msg.reader().readByte();
		if (!BoardScr.setR_B(b, b2))
		{
			return;
		}
		switch (msg.command)
		{
		case 20:
		{
			sbyte b3 = msg.reader().readByte();
			BCBoardScr.me.saveTime = b3;
			BCBoardScr.me.onStartGame(b, b2, b3);
			break;
		}
		case 21:
		{
			sbyte b4 = msg.reader().readByte();
			if (b4 == -1)
			{
				BCBoardScr.me.setBotCmd();
				BCBoardScr.me.canpointer = false;
			}
			else if (b4 != -1)
			{
				for (int k = 0; k < 6; k++)
				{
					BCBoardScr.me.moneySV[b4][k] = msg.reader().readByte();
				}
				BCBoardScr.me.onMove(b4);
			}
			break;
		}
		case 37:
		{
			sbyte[] array2 = new sbyte[3];
			for (int m = 0; m < 3; m++)
			{
				array2[m] = msg.reader().readByte();
			}
			BCBoardScr.me.onResult(array2);
			BCBoardScr.me.setCmdWaiting();
			break;
		}
		case 100:
		{
			sbyte seatI = msg.reader().readByte();
			BCBoardScr.me.onSetTurn(seatI);
			break;
		}
		case 65:
		{
			sbyte b7 = msg.reader().readByte();
			sbyte b8 = msg.reader().readByte();
			sbyte b9 = msg.reader().readByte();
			sbyte b10 = msg.reader().readByte();
			if (b8 != b9 && b10 > 0)
			{
				BCBoardScr.me.moneySV[b7][b9] = b10;
				BCBoardScr.me.onHaphom(b7, b8, b9);
			}
			break;
		}
		case 60:
		{
			sbyte b5 = 0;
			sbyte b6 = 0;
			b5 = msg.reader().readByte();
			b6 = msg.reader().readByte();
			int moneyValue = msg.reader().readInt();
			BCBoardScr.me.onSetPlayer(b5, b6, moneyValue);
			break;
		}
		case 51:
		{
			int[] array = new int[BoardScr.avatarInfos.size()];
			for (int l = 0; l < array.Length; l++)
			{
				array[l] = msg.reader().readInt();
			}
			BCBoardScr.me.onFinish(array);
			break;
		}
		case 62:
		{
			sbyte b3 = msg.reader().readByte();
			BCBoardScr.me.saveTime = b3;
			for (int i = 0; i < BoardScr.avatarInfos.size(); i++)
			{
				for (int j = 0; j < 6; j++)
				{
					BCBoardScr.me.moneySV[i][j] = msg.reader().readByte();
				}
			}
			BCBoardScr.me.onPlaying(b3);
			break;
		}
		}
	}
}

File: client/java/src/avt/BaucuaMsgHandler.java
package avt;

public final class BaucuaMsgHandler implements IMiniGameMsgHandler {
   public static BaucuaMsgHandler instance;

   public final void onMessage(Message var1) {
      try {
         int var2 = var1.reader().readByte();
         int var3 = var1.reader().readByte();
         if (!BoardScr.setR_B((byte)var2, (byte)var3)) {
            return;
         }

         int var6;
         byte var7;
         switch (var1.command) {
            case 20:
               var7 = var1.reader().readByte();
               BCBoardScr.me_.saveTime = var7;
               BCBoardScr.me_.onStartGame((byte)var2, (byte)var3, var7);
               return;
            case 21:
               if ((var7 = var1.reader().readByte()) == -1) {
                  BCBoardScr.me_.setBotCmd();
                  BCBoardScr.me_.canpointer = false;
                  return;
               }

               if (var7 != -1) {
                  for(var3 = 0; var3 < 6; ++var3) {
                     BCBoardScr.me_.moneySV[var7][var3] = var1.reader().readByte();
                  }

                  BCBoardScr.me_.onMove(var7);
                  return;
               }
               break;
            case 37:
               byte[] var10 = new byte[3];

               for(var2 = 0; var2 < 3; ++var2) {
                  var10[var2] = var1.reader().readByte();
               }

               BCBoardScr.me_.onResult(var10);
               BoardScr.setCmdWaiting();
               return;
            case 51:
               int[] var8 = new int[BoardScr.avatarInfos.size()];

               for(var3 = 0; var3 < var8.length; ++var3) {
                  var8[var3] = var1.reader().readInt();
               }

               BCBoardScr.me_.onFinish(var8);
               return;
            case 60:
               var7 = var1.reader().readByte();
               byte var9 = var1.reader().readByte();
               var6 = var1.reader().readInt();
               BCBoardScr.me_.showFlyText5Baucua(var7, var9, var6);
               return;
            case 62:
               var2 = var1.reader().readByte();
               BCBoardScr.me_.saveTime = (byte)var2;

               for(var3 = 0; var3 < BoardScr.avatarInfos.size(); ++var3) {
                  for(var2 = 0; var2 < 6; ++var2) {
                     BCBoardScr.me_.moneySV[var3][var2] = var1.reader().readByte();
                  }
               }

               BCBoardScr.me_.onPlaying();
            default:
               return;
            case 65:
               var2 = var1.reader().readByte();
               var3 = var1.reader().readByte();
               byte var4 = var1.reader().readByte();
               var6 = var1.reader().readByte();
               if (var3 != var4 && var6 > 0) {
                  BCBoardScr.me_.moneySV[var2][var4] = (byte)var6;
                  BCBoardScr.me_.onHaphom((byte)var2, (byte)var3, var4);
                  return;
               }
               break;
            case 100:
               var2 = var1.reader().readByte();
               BCBoardScr.me_.onSetTurn((byte)var2);
               return;
         }
      } catch (Exception var10) {
         var10.printStackTrace();
      }

   }
}
